README-new.txt

```diff
@@ -0,0 +1,209 @@
+= The language
+
+Modula-3 is an optionally safe systems programming language.
+ It has garbage collection (for the optionally safe part).
+ Extent implementations are compiled statically to native code.
```

Comment:
The word Extent is hard to read. Existing?

Reply:
I guess the word you want there is "Extant", but the whole thing could possibly be reworded as

> Existing implementation compile ahead-of-time to native code.

(Because apparently what compilers have done for decades is now a special thing with its own lingo.)

Reply:
1. "ahead of time" ugh. It is called "compiling". I won't say much right about this termonology I am tired and grumpy. :)
2. I don't like the current README. You can see mine is based on it, but I don't like that it points people to the.sh files and I like that I at least tried to explain various ways to run them (.py and .sh do about the same). Maybe allude to the .sh files for the unusual systems without .py.


Aside not for docs:
 I have py2 on OSF1.
 I have py3 but not current on OSF1.
 Porting to py3 is challenging.
 I would like to try bringup on DOS maybe.
 Point is though. py2 and perhaps py3 are pretty darn portable, the space that needs .sh is pretty small imho.
  We could offer py -fake that writes out the commands, perhaps.

Or has been said many times, rewrite the scripts in quake or mod3, but I don't know, I am torn.

Reply:
The scripts have definitely been a point of confusion for me.  I recognize that the python scripts are there for portability, but I'm so used to shell that they're a bit foreign.

I like the idea of rewriting them in quake, but I don't know enough yet to know how practical that is.  Of course there would still have to be some sort of bootstrap process.

I saw somewhere quake being compared to scons.  I don't know scons, but the scripts always remind me of cmake, which gives me the idea of organizing/modularizing them in the same style.  But I'm getting in way over my head.

But yes it's a mess, there will always be tension between "it works for me" and "it needs to be portable", and some sort of native/built-in solution may be the best way to resolve that.

For now just cataloguing what's there is a help.

And I get the grumpy.  :)

Reply:
We "have py2 on" on DOS , but we do not have py3 ( I found working py2, but do not found py3)